### PR TITLE
[RAIN-2934] : NLP Code cleanup

### PR DESCRIPTION
**Pull request by Sarvesh-eGov**

### CODE REVIEW

File: build/build-config.yml [hunk 1]
```diff
@@ -242,16 +242,22 @@ config:
...
       - work-dir: "libraries/enc-client"
         image-name: "enc-client"
+<<<<<<< HEAD
+=======
         
   - name: "builds/core-services/libraries/tracer"
     build:
       - work-dir: "libraries/tracer"
         image-name: "tracer"
+>>>>>>> develop
```
Comment (talele08) [inline build/build-config.yml:252]: resolve this at all places

**Review by talele08 — CHANGES_REQUESTED** [review]

File: build/build-config.yml (hunk 1 above)
Comment (Sarvesh-eGov) [reply build/build-config.yml]: Done

File: nlp-engine/app_vf.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+    payload= default
+    
+    #inp=translator.translate(inp,dest='en')
+    #src1=inp.src
+    #inp=inp.text
+    inp=inp.lower()
+    
+
+    url = "https://api.gupshup.io/sm/api/v1/msg"
```
Comment (talele08) [inline nlp-engine/app_vf.py:68]: don't use hard coded values. Make it configurable

File: nlp-engine/app_vf.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+      'apikey': '37cef3c8bf164df7cdc0a36eae94beec',
+      'cache-control': 'no-cache'
+                      }
+
+    if len(inp.split())==1 and inp!="hi" and inp!="hello":
+        answer=find_city(inp)[0].upper()
+        answer=answer[0]+answer[1:].lower()
+
+        k=payload.index("&message")
+        payload=payload[0:k]+"&message="+translator.translate("We have detected *"+answer+"* as your city.",dest='en').text+ "&src.name=chatbotpayment"
```
Comment (talele08) [inline nlp-engine/app_vf.py:81]: Hardcoding of text. Fetch it from localization. Take care of it at all the places

File: nlp-engine/auto_trial.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+#for i in range(0,len(stopwords)):
+#    stopwords[i]= stopwords[i][0:len(stopwords[i])-1].lower()
+
+rem=stopwords
+#rem= rem+['show','see']
+keep=['are','to','be','not']
+from autocorrect import Speller
+spell=Speller(lang='en')
+
+r= requests.get('https://raw.githubusercontent.com/egovernments/egov-mdms-data/DEV/data/pb/Chatbot/KeyWords.json')
```
Comment (talele08) [inline nlp-engine/auto_trial.py:23]: Don't hardcode, the link is only for dev env

File: nlp-engine/city_extract.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+from googletrans import Translator
+from nltk.util import ngrams
+import requests
+import json
+
+translator= Translator()
+
+punct= string.punctuation
+
+r= requests.get('https://raw.githubusercontent.com/egovernments/egov-mdms-data/DEV/data/pb/Chatbot/CityNames.json')
```
Comment (talele08) [inline nlp-engine/city_extract.py:13]: hardcoding

File: nlp-engine/app_vf.py [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+        elif 'trade' in c:
+            payload = 'channel=whatsapp&source=917834811114&destination='+destination+'&message=%7B%22type%22%3A%22file%22%2C%22url%22%3A%22https%3A//www.buildquickbots.com/whatsapp/media/sample/pdf/sample01.pdf%22%2C%22caption%22%3A%22%22%2C%22filename%22%3A%22Trade_license_receipts.pdf%22%7D&src.name=chatbotpayment'
+
+        elif 'property' in c:
+            payload = 'channel=whatsapp&source=917834811114&destination='+destination+'&message=%7B%22type%22%3A%22file%22%2C%22url%22%3A%22https%3A//www.buildquickbots.com/whatsapp/media/sample/pdf/sample01.pdf%22%2C%22caption%22%3A%22%22%2C%22filename%22%3A%22Property_Tax_receipts.pdf%22%7D&src.name=chatbotpayment'
+
+        else:
+            
+            k=payload.index("&message")
```
Comment (talele08) [inline nlp-engine/app_vf.py:123]: add comments in code to explain what is happening in the code

**Review by talele08 — CHANGES_REQUESTED** [review]

File: nlp-engine/app_vf.py (hunk 2 above)
Comment (Sarvesh-eGov) [reply nlp-engine/app_vf.py:68]: This file is renamed as ChatbotApi.py in the src. Can u plz check that?

File: nlp-engine/auto_trial.py (hunk 4 above)
Comment (Sarvesh-eGov) [reply nlp-engine/auto_trial.py:23]: This file is renamed as Preprocessing.py in src. Please check that.

File: nlp-engine/app_vf.py (hunk 6 above)
Comment (Sarvesh-eGov) [reply nlp-engine/app_vf.py:123]: Done

File: nlp-engine/city_extract.py (hunk 5 above)
Comment (Sarvesh-eGov) [reply nlp-engine/city_extract.py:13]: This file is renamed as CityExtract.py in src. Please check that.

File: nlp-engine/src/ChatbotApi.py [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+        
+
+        response = requests.request("POST", url, headers=headers, data = payload)
+
+        k=payload.index(MESSAGE_TOKEN)
+        payload=payload[0:k]+MESSAGE_TOKEN+translator.translate(result,dest=sourceLanguage).text+ SRC_NAME
+
+        
+
+
```
Comment (talele08) [inline nlp-engine/src/ChatbotApi.py:159]: Please do proper formatting

File: nlp-engine/src/Config.py [hunk 8]
```diff
@@ -11,4 +38,64 @@
...
+WATER = 'water'
+TRADE = 'trade'
+PROPERTY = 'property'
+
+RECEIPT_TOKEN = 'Showing'
+BILL_TOKEN = 'Visit'
+BILL_TOKEN_NEW='You'
+UTF_8 = 'utf-8'
+
+ASK_LOCALITY_NAME = "Type the name of your locality\nType 'mseva' to return back to the main menu."
```
Comment (talele08) [inline nlp-engine/src/Config.py:79]: Move this to localization service, not in configuration

**Review by talele08 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**